types/src/on_chain_config.rs

```diff
@@ -9,7 +9,38 @@ use crate::{
 };
 use libra_crypto::HashValue;
 use move_core_types::identifier::Identifier;
-use serde::{Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+/// Trait to be implemented by a storage type from which to read on-chain configs
+#[async_trait::async_trait(?Send)]
+pub trait ConfigStorage {
+    async fn fetch_config(&self, access_path: AccessPath) -> Option<Vec<u8>>;
+}
+
+/// Trait to be implemented by a Rust struct representation of an on-chain config
+/// that is stored in storage as a deserialized byte array
+#[async_trait::async_trait(?Send)]
+pub trait OnChainConfig {
+    const IDENTIFIER: &'static str;
```

Comment:
nit: I think it will be handy to have enum for all available on-chain configs. But we can delay that for future PR where we'll try to aggregate them in single Rust object that will be passed around

Reply:
this looks nice and can be extended to not only config but any move data with some effort cc: @davidiw @sblackshear

Reply:
Note: I'm realizing this pretty big limitation exists though: https://doc.rust-lang.org/error-index.html#E0562 - you can't make trait objects for traits with associated constants 😞